sycl/test-e2e/Matrix/get_coord_int8_matA_impl.hpp

```diff
@@ -97,47 +99,78 @@ void matrix_sum_rows(big_matrix<T, M, K> &A) {
            const auto sg_starty = global_idy - spmd_item.get_local_id(1);
 
            sycl::sub_group sg = spmd_item.get_sub_group();
-           joint_matrix<sub_group, int8_t, use::a, TM, TK, layout::row_major>
+           joint_matrix<sub_group, T, use::a, TileRows, TileCols,
+                        layout::row_major>
                sub_a;
            joint_matrix_load(
                sg, sub_a,
                accA.template get_multi_ptr<access::decorated::no>() +
-                   (sg_startx * TM * K) + sg_starty / sg_size * TK,
-               K);
+                   (sg_startx * TileRows * Cols) +
+                   sg_starty / sg_size * TileCols,
+               Cols);
 
-           int32_t sum_local_rows[M] = {0};
+           TResult sum_local_rows[Rows] = {0};
 
            ext::intel::experimental::matrix::joint_matrix_apply(
-               sg, sub_a, [&](int8_t &x, size_t row, size_t col) {
-                 sum_local_rows[row + global_idx * TM] += x;
+               sg, sub_a, [&](T &x, size_t row, size_t col) {
+                 sum_local_rows[row + global_idx * TileRows] += x;
                });
-           for (int i = 0; i < M; ++i) {
+           for (int i = 0; i < Rows; ++i) {
              sum_local_rows[i] =
                  reduce_over_group(sg, sum_local_rows[i], sycl::plus<>());
 
              // only Groups leader performs the global reduction
-             if (global_idy % sg_size == 0)
-               atomic_fetch_add(v[i], sum_local_rows[i]);
+             if (global_idy % sg_size == 0) {
+               sycl::atomic_ref<TResult, sycl::memory_order::relaxed,
+                                sycl::memory_scope::device>
+                   aref(v[i]);
+               aref.fetch_add(sum_local_rows[i]);
```

Comment:
Why did you replace atomic_fetch_add with atomic_ref + fetch_add? atomic_fetch_add seems to be less verbose.

Reply:
From this declaration `auto v = sum_rows_v.get_access<access::mode::atomic>(cgh)`, `access::mode::atomic` has been deprecated. Using `atomic_ref` is the current best practice performing atomic operations on memory. I have corroborated this with Steffen Larsen.

Reply:
Thanks for information